File: day1/solution.py
import math

def fuel_required_for_module(mass):
  fuelreq = math.floor(mass / 3) - 2
  return fuelreq

def fuel_required_for_module_part2(mass):
  last = fuel_required_for_module(mass)
  total = 0
  while last > 0:
    total += last
    last = fuel_required_for_module(last)
  return total



masses = []
f = open("input.txt", "r")
for m in f:
  masses.append(int(m))

# Part One
fuels_one = list(map(fuel_required_for_module, masses))
total_fuel_one = sum(fuels_one)
print("Part One: total fuel required: %d"% (total_fuel_one))

# Part Two

fuels_two = list(map(fuel_required_for_module_part2, masses))
total_fuel_two = sum(fuels_two)
print("Part Two: total fuel required: %d"% (total_fuel_two))

File: day1/solution.rb
def fuel_required_for_module(mass)
  fuelreq = (mass.to_i / 3) - 2
  fuelreq
end

masses = File.open("input.txt", "r").readlines.map(&:to_i)
fuels = masses.map{ |m| fuel_required_for_module(m) }
total_fuel = fuels.sum

puts "total fuel required: #{total_fuel}"
